src/deepsparse/pipeline_loggers/multi_logger.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A manager that oversees all the pipeline loggers
+"""
+
+from typing import Any, Dict, List, Union
+
+from deepsparse.pipeline_loggers.pipeline_logger import PipelineLogger
+from deepsparse.pipeline_loggers.prometheus_pipeline_logger import PrometheusLogger
+from deepsparse.timing.timing_schema import InferenceTimingSchema
+
+
+__all__ = ["MultiLogger"]
+
+LOGGERS = [PrometheusLogger]
+SUPPORTED_LOGGERS = {logger.identifier: logger for logger in LOGGERS}
+
+
+class MultiLogger(PipelineLogger):
+    """
+    Object that contains multiple loggers for
+    the given inference pipeline.
+
+    The envisioned lifecycle of a logger
+    manager:
+
+    ```
+    pipeline = ... # define a pipeline
+    pipeline_name = pipeline.name # fetch the pipeline name
+
+    # create a MultiLogger
+    logger_manager = MultiLogger("logger_name")
+    or
+    logger_manager = MultiLogger(["logger_name_1", "logger_name_2", ...])
+
+    # log the data for the particular inference pipeline
+
+    timings, data = ...
+    logger.log_timings(pipeline_name, timings)
+    logger.log_data(pipeline_name, data)
+
+    ```
+
+    :param logger_identifiers: The identifier(s) of the logger types that
+        will be created within the scope of the manager. This can be either a
+        single identifier, or a list of identifiers (for multiple loggers).
+    """
+
+    def __init__(
+        self,
+        logger_identifiers: Union[str, List[str]],
+    ):
```

Comment:
We could just pass in the list of instantiated logger objects directly into this class `loggers: List[PipelineLogger]`, and move the creation/parsing logic (`self._setup`, `self._start`) outside of this class to somewhere else (the server I guess?)

Then this class would be a really tightly scoped util class to easily wrap multiple loggers. It'd be easier to test to since you won't have to go through strings and could just pass in the loggers you want to use.

Reply:
I like this idea a lot!